disttask/framework/framework_test.go

```diff
@@ -696,18 +695,22 @@ func TestGC(t *testing.T) {
 	mgr, err := storage.GetTaskManager()
 	require.NoError(t, err)
 
-	// check the subtask history table
-	historySubTasksCnt, err := storage.GetSubtasksFromHistoryForTest(mgr)
-	require.NoError(t, err)
-	require.Equal(t, 4, historySubTasksCnt)
-
-	// wait for gc
-	time.Sleep(10 * time.Second)
+	var historySubTasksCnt int
+	require.Eventually(t, func() bool {
+		historySubTasksCnt, err = storage.GetSubtasksFromHistoryForTest(mgr)
+		if err != nil {
+			return false
+		}
+		return historySubTasksCnt > 0
```

Comment:
```suggestion
		return historySubTasksCnt ==4
```

Reply:
this won't fix the issue. it might have already GCed before this eventually

Reply:
L686 set the `historySubtaskTableGcInterval` to 10s, and this Eventually interval is 500*time.Millisecond, so it is very small probability to encounter your situation.

Reply:
> very small probable

that's what we see in the issue

Reply:
Seem what I can do is reduce the eventual interval and decrease the probability. They are asynchronous.

Reply:
And add subtaskHistoryKeepSeconds

Reply:
Increasing `subtaskHistoryKeepSeconds` is a way to reduce the probability, What do you think?

Reply:
> Increasing `subtaskHistoryKeepSeconds` is a way to reduce the probability, What do you think?

No, why not add a failpoint to ensure the check subtask cnt == 4 succeed, then start gc?